spec/classes/puppet_agent_spec.rb

```diff
@@ -237,7 +237,7 @@ def global_facts(facts, os)
         before(:each) do
           Puppet::Parser::Functions.newfunction(:pe_build_version, type: :rvalue) { |_args| '2000.0.0' }
           Puppet::Parser::Functions.newfunction(:pe_compiling_server_aio_build, type: :rvalue) { |_args| '1.10.100' }
-          Puppet::Parser::Functions.newfunction(:pe_compiling_server_version, type: :rvalue) { |_args| '2.20.200' }
+          Puppet::Parser::Functions.newfunction(:defined, type: :rvalue) { |_args| true }
```

Comment:
Mocking `defined` to return true might break things. According to the tests I did with the module it *seems* to be fine, but it feel fragile :fearful:.

Reply:
I'm not sure I understand why this is getting mocked here; @bastelfreak can you shed some light?